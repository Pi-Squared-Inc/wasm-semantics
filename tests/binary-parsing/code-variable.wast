(module $mymodule
  (func (param i32) (result i32)
    (local i32 i64)
    local.get 0
    local.set 0
    local.get 1
    local.set 1
    local.get 0
    local.tee 0
    drop
    (return (i32.const 1))
  )
)
