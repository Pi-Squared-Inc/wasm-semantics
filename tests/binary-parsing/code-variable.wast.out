<generatedTop>
  <k>
    #module (... types: #type (... type: [ i32  .ValTypes ] -> [ i32  .ValTypes ] , metadata:  )  .EmptyStmts , funcs: #func (... type: 0 , locals: [ i32  i64  .ValTypes ] , body: #local.get ( 0 )  #local.set ( 0 )  #local.get ( 1 )  #local.set ( 1 )  #local.get ( 0 )  #local.tee ( 0 )  drop  i32 . const 1  return  .EmptyStmts , metadata: #meta (... id:  , localIds: .Map ) )  .EmptyStmts , tables: .EmptyStmts , mems: .EmptyStmts , globals: .EmptyStmts , elem: .EmptyStmts , data: .EmptyStmts , start: .EmptyStmts , importDefns: .EmptyStmts , exports: .EmptyStmts , metadata: #meta (... id:  , funcIds: .Map , filename: .String ) ) ~> .K
  </k>
  <wasm>
    <instrs>
      .K
    </instrs>
    <valstack>
      .ValStack
    </valstack>
    <curFrame>
      <locals>
        .List
      </locals>
      <curModIdx>
        .Int
      </curModIdx>
    </curFrame>
    <moduleRegistry>
      .Map
    </moduleRegistry>
    <moduleIds>
      .Map
    </moduleIds>
    <moduleInstances>
      .ModuleInstCellMap
    </moduleInstances>
    <nextModuleIdx>
      0
    </nextModuleIdx>
    <mainStore>
      <funcs>
        .FuncDefCellMap
      </funcs>
      <nextFuncAddr>
        0
      </nextFuncAddr>
      <tabs>
        .TabInstCellMap
      </tabs>
      <nextTabAddr>
        0
      </nextTabAddr>
      <mems>
        .List
      </mems>
      <globals>
        .GlobalInstCellMap
      </globals>
      <nextGlobAddr>
        0
      </nextGlobAddr>
      <elems>
        .ElemInstCellMap
      </elems>
      <nextElemAddr>
        0
      </nextElemAddr>
    </mainStore>
    <deterministicMemoryGrowth>
      true
    </deterministicMemoryGrowth>
  </wasm>
</generatedTop>
